app/components/upload-container.tsx

```diff
@@ -104,20 +108,20 @@ const RecordAudio = ({
             if (audioRecorder.current === null) {
               return
             }
-            if (recordingInfo.value.isRecording) {
-              audioRecorder.current.stopRecording()
+            if (isRecording) {
+              audioRecorder.current.stopRecording(setIsRecording)
             } else {
-              audioRecorder.current.startRecording()
+              audioRecorder.current.startRecording({setIsRecording, setNormalizedLoudness, setRecordedAudioDurationSecs, isRecording, normalizedLoudness})
             }
           }}>
-          {recordingInfo.value.isRecording ? <StopIcon /> : <MicrophoneIcon />}
+          {isRecording ? <StopIcon /> : <MicrophoneIcon />}
         </Button>
         <Button variant="outline" size="icon" onClick={onClose}>
           <XMarkIcon />
         </Button>
       </div>
-      <MicrophoneVolumeIndicator className="mb-3" />
-      <MicrophoneRecordTimeIndicator />
+      <MicrophoneVolumeIndicator className="mb-3" isRecording={isRecording} normalizedLoudness={normalizedLoudness} />
+      <MicrophoneRecordTimeIndicator recordedAudioDurationSecs={recordedAudioDurationSecs}/>
```

Comment:
every time we have same same name props/variables i like to pass them like this. Reducing boilerplate and duplication

```suggestion
      <MicrophoneVolumeIndicator className="mb-3" {...{isRecording, normalizedLoudness}} />
      <MicrophoneRecordTimeIndicator {...{recordedAudioDurationSecs}} />
```